emulate.c

```diff
@@ -651,7 +653,7 @@ static uint32_t csr_csrrs(struct riscv_t *rv, uint32_t csr, uint32_t val)
     if (!c)
         return 0;
 
-    const uint32_t out = *c;
+    const uint32_t out = (csr == CSR_FFLAGS) ? *c & FFLAG_MASK : *c;
```

Comment:
Rewrite as following:
```c
uint32_t out = *c;
if (csr == CSR_FFLAGS)
    out &= FFLAG_MASK;
```

Reply:
I rewrote the function and also add the same code in `csr_csrrw` and `csr_csrrc`.

Since FCSR is defined only if ENABLE_RV32F is defined:
```
static uint32_t *csr_get_ptr(struct riscv_t *rv, uint32_t csr)
{
    switch (csr) {
    case CSR_CYCLE:
        return (uint32_t *) (&rv->csr_cycle) + 0;
    case CSR_CYCLEH:
        return (uint32_t *) (&rv->csr_cycle) + 1;
    case CSR_MSTATUS:
        return (uint32_t *) (&rv->csr_mstatus);
    case CSR_MTVEC:
        return (uint32_t *) (&rv->csr_mtvec);
    case CSR_MISA:
        return (uint32_t *) (&rv->csr_misa);
    case CSR_MSCRATCH:
        return (uint32_t *) (&rv->csr_mscratch);
    case CSR_MEPC:
        return (uint32_t *) (&rv->csr_mepc);
    case CSR_MCAUSE:
        return (uint32_t *) (&rv->csr_mcause);
    case CSR_MTVAL:
        return (uint32_t *) (&rv->csr_mtval);
    case CSR_MIP:
        return (uint32_t *) (&rv->csr_mip);
#ifdef ENABLE_RV32F
    case CSR_FFLAGS:
        return (uint32_t *) (&rv->csr_fcsr);
    case CSR_FCSR:
        return (uint32_t *) (&rv->csr_fcsr);
#endif
    default:
        return NULL;
    }
}
```

Should I also use macro inside `csr_csrrs` ?

Reply:
> Should I also use macro inside `csr_csrrs` ?

You can do it in another pull request.